scaleway-core/scaleway_core/client.py

```diff
@@ -19,6 +18,11 @@
 class Client(Profile):
     _request_count: int = 1
 
+    """
+    session is used to share a session between multiple API clients.
+    """
+    session: Optional[requests.Session] = None
```

Comment:
I'm really not sure how to make an elegant setter for this :/ Sorry. 

Personally, I always use `Client.from_config_file_and_env()` to create my client. I feel like adding a session named argument here would expose an "advanced" setting in a function that should be kept clean. Maybe I could use a builder pattern of some sort? But it seems way too overengineered.  For now, I'm just adding the attribute which can always be set.